File: FutronicSDK/Examples/Net/Vs2013/WorkedEx/SelectUser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.IO;
using System.Windows.Forms;

namespace Futronic.SDK.WorkedEx
{
    public partial class SelectUser : Form
    {
        private List<DbRecord> m_Users;
        private int m_SelectedIndex;

        public SelectUser( List<DbRecord> Users, String szDbDir )
        {
            InitializeComponent();
            txtDatabaseDir.Text = szDbDir;
            m_Users = Users;
            for( int i = 0; i < m_Users.Count; i++ )
            {
                lstUsers.Items.Add(m_Users[i].UserName);
            }
            lstUsers.SelectedIndex = 0;
            m_SelectedIndex = -1;
        }

        public DbRecord SelectedUser
        {
            get
            {
                if (m_Users.Count == 0 || m_SelectedIndex == -1)
                    return null;
                return m_Users[m_SelectedIndex];
            }
        }

        private void btnSelect_Click(object sender, EventArgs e)
        {
            m_SelectedIndex = lstUsers.SelectedIndex;
        }
    }
}

File: FutronicSDK/Examples/Net/Vs2013/WorkedEx.VB/SelectUser.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.IO
Imports System.Windows.Forms

Namespace Futronic.SDK.WorkedEx
	Public Partial Class SelectUser
		Inherits Form
		Private m_Users As List(Of DbRecord)
		Private m_SelectedIndex As Integer

		Public Sub New(ByVal Users As List(Of DbRecord), ByVal szDbDir As String)
			InitializeComponent()
			txtDatabaseDir.Text = szDbDir
			m_Users = Users
			Dim i As Integer = 0
			Do While i < m_Users.Count
				lstUsers.Items.Add(m_Users(i).UserName)
				i += 1
			Loop
			lstUsers.SelectedIndex = 0
			m_SelectedIndex = -1
		End Sub

		Public ReadOnly Property SelectedUser() As DbRecord
			Get
				If m_Users.Count = 0 OrElse m_SelectedIndex = -1 Then
					Return Nothing
				End If
				Return m_Users(m_SelectedIndex)
			End Get
		End Property

		Private Sub btnSelect_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSelect.Click
			m_SelectedIndex = lstUsers.SelectedIndex
		End Sub
	End Class
End Namespace